components/directory/details-form.tsx

```
@@ -0,0 +1,495 @@
+    });
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    onSubmit(formData);
+  };
+
+  const getStepNumber = () => selectedPlan === "free" ? 1 : 2;
+  const requiredFieldsCount = 4; // name, link, category, description
+  const completedRequiredFields = ['name', 'link', 'category', 'description'].filter(field => 
+    formData[field] && formData[field].toString().trim()
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)